homeassistant/components/version/const.py

```diff
@@ -61,8 +61,6 @@
 
 BOARD_MAP: Final[dict[str, str]] = {
     "OVA": "ova",
-    "RaspberryPi": "rpi",
-    "RaspberryPi Zero-W": "rpi0-w",
```

Comment:
There are existing config entries that have a configured board that loads from this map unconditionally.

Removing entries here, will break existing config entries. I guess we need to either migrate or at least prevent crashes and add a proper error message (right now it will crash with a stack trace).

Reply:
Hm, I see. Yeah this came up due to a existing conifg entry, see https://github.com/home-assistant/operating-system/issues/1914.

Migrate: I guess the question would be where to :sweat_smile: For `intel-nuc` we could do it, but not for the now unsupported boards. I guess a warning would make sense there, mentioning that the board does no longer get any updates.

Reply:
IMO we should remove entries that are no longer valid with a proper breaking-change, and make sure that the integration also states that it's no longer valid if loaded on startup.